recipes/pugixml/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+        "header_only": [True, False],
+        "wchar_mode": [True, False],
+        "no_exceptions": [True, False]
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'header_only': False,
+        'wchar_mode': False,
+        'no_exceptions': False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.header_only:
+            if self.settings.os != 'Windows':
+                del self.options.fPIC
+            del self.options.shared
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTS"] = False
+        if self.settings.os == 'Windows' and self.settings.compiler == 'Visual Studio':
+            self._cmake.definitions['CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS'] = self.options.shared
+        self._cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+        return self._cmake
+
+    def build(self):
+        header_file = os.path.join(self._source_subfolder, "src", "pugiconfig.hpp")
+        if self.options.wchar_mode:
+            tools.replace_in_file(header_file, "// #define PUGIXML_WCHAR_MODE", '''#define PUGIXML_WCHAR_MODE''')
+        if self.options.no_exceptions:
+            tools.replace_in_file(header_file, "// #define PUGIXML_NO_EXCEPTIONS", '''#define PUGIXML_NO_EXCEPTIONS''')
+
+        if self.options.header_only:
+            tools.replace_in_file(header_file, "// #define PUGIXML_HEADER_ONLY", '''#define PUGIXML_HEADER_ONLY''')
+        else:
+            cmake = self._configure_cmake()
+            cmake.build()
+
+    def package(self):
+        readme_contents = tools.load(os.path.join(self._source_subfolder, "readme.txt"))
+        license_contents = readme_contents[readme_contents.find("This library is"):]
+        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), license_contents)
+        if self.options.header_only:
+            source_dir = os.path.join(self._source_subfolder, "src")
+            self.copy(pattern="*", dst="include", src=source_dir)
+        else:
+            cmake = self._configure_cmake()
+            cmake.install()
+            tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
+            tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+
+    def package_id(self):
+        if self.options.header_only:
+            self.info.header_only()
```

Comment:
I think this is incorrect because `header_only` will clear `options`, `settings` and `requires`.
See https://github.com/conan-io/conan/blob/0f1bd5a75fcc40b73e09a19797094b9ee3a3c234/conans/model/info.py#L560-L563
You don't want to overwrite a package where e.g. the option `wcher_mode` is different.
```suggestion
            self.info.settings.clear()
            self.info.requires.clear()
```

Reply:
@madebr , thank you for link with details of work header_only() method, but seems it does not important for this case, because method build(self) executes earlier and makes all needed changes in the "pugiconfig.hpp". I've tested combination of options "header-only", "wchar_mode" and "no_exceptions" and all work well (in last commit I've added displaying these options from test package).
```
conan create . pugixml/1.10@  -o *:header_only=True -o *:wchar_mode=True -o *:no_exceptions=True
```

Reply:
So building pugixml/1.10@ with:

1.  `-o *:header_only=True -o *:wchar_mode=False`
2. `-o *:header_only=True -o *:wchar_mode=True`

will create 2 packages?
I thought, because `options` was cleared, that these were aliased to the same `package_id`.

Reply:
Got it! Thank you for explanation, I've committed the suggested change.